api/document/serializers/content.py

```diff
@@ -123,19 +127,34 @@ def get_content_type(self, instance: NestableAnnotation):
         return self.CONTENT_TYPE
 
     def get_inlines(self, instance: NestableAnnotation):
+        if self.IS_LEAF:
+            return []
         return NestedAnnotationSerializer(
             instance.children, context=self.context, many=True).data
 
     def get_text(self, instance: Annotation):
         return self.doc_node_text[instance.start:instance.end]
 
+    def to_internal_value(self, data: JsonDict) -> JsonDict:
+        inlines = [] if self.IS_LEAF else data.get('inlines', [])
```

Comment:
Note that this'll silently drop bad user data (a __text__ with non-empty `inlines`), which seems risky. Should it be raising a validation exception instead?